p2p/peer.go

```diff
@@ -310,28 +291,52 @@ func (p *peer) OnStart() error {
 		return err
 	}
 
```

Comment:
Curious, do we need to explicitly call `BaseService.OnStart()`?

Reply:
nope